### PR TITLE
Fix laravel transaction naming (issue #86) - Don't use getName() if it's a auto generated route name

### PR DESCRIPTION
**Pull request by stockalexander**
Route::getName() returns a random generated string (generated::*) if a route isn't named [(Named Routes)](https://laravel.com/docs/8.x/routing#named-routes) and routes are cached via `php artisan route:cache` since laravel 7.x.

This causes the problem mentioned in #86. To fix the problem, I added a check if the returned string begins with `generated::`, to skip it and use `Route::getAction()` instead.

### CODE REVIEW

**Comment by Fahmy-Mohammed** [comment]
ok jenkins

**Review by Fahmy-Mohammed — DISMISSED** [review]
Thanks for adding this fix @stockalexander. I've tested this locally and it works well! This should be included in the upcoming PHP agent release (version 9.18).

**Comment by Fahmy-Mohammed** [comment]
@stockalexander before we can merge this, would you be able to change this to the dev branch instead of main? That is where all of the work included in the upcoming release is contained.

**Comment by stockalexander** [comment]
@Fahmy-Mohammed: Nice! I rebased my branch to the dev branch (+ force pushed it) and updated the PR. Can you merge it now or should I change something else?

**Comment by Fahmy-Mohammed** [comment]
@stockalexander There are some internal Jenkins builds that are running currently, so I'll merge this once those tests pass. Awesome work identifying and adding the fix. We really appreciate it!

**Comment by Fahmy-Mohammed** [comment]
ok jenkins

**Review by Fahmy-Mohammed — APPROVED** [review]

**Comment by Fahmy-Mohammed** [comment]
ok jenkins

**Pull request merged**
